### PR TITLE
fix(upgrade): improve logs and pre-upgrade checks

### PR DESCRIPTION
**Pull request by shubham14bajpai**
Signed-off-by: shubham <shubham.bajpai@mayadata.io>

**What this PR does / why we need it**:
This PR improves the error reporting and pre-flight checks for the upgrade. The mismatch of `errors` package import was causing loss of error messages. This PR makes the import consistent for all files.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: cmd/upgrade/executor/setup_job.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ import (
 
 	//"k8s.io/klog"
 	"github.com/spf13/cobra"
-	//errors "github.com/openebs/maya/pkg/errors/v1alpha1"
```
Comment (sagarkrsd) [inline cmd/upgrade/executor/setup_job.go]: Please remove the imports which are not being used here.

Comment (shubham14bajpai) [reply cmd/upgrade/executor/setup_job.go]: Will remove this in upcoming PRs.

File: pkg/upgrade/upgrader/helper.go [hunk 2]
```diff
@@ -307,3 +311,10 @@ func buildUpgradeTask(kind, name, openebsNamespace string) *apis.UpgradeTask {
 	}
 	return utaskObj
 }
+
+func verifyDeploymentReplicaStatus(d *appsv1.Deployment) error {
+	if d.Status.ReadyReplicas != *d.Spec.Replicas {
+		return errors.New(d.Name + " deployment pod is not in running state.")
```
Comment (vishnuitta) [inline pkg/upgrade/upgrader/helper.go]: better to give more info like ReadyReplicasCount and Spec.Replicas as well

Comment (vishnuitta) [reply pkg/upgrade/upgrader/helper.go]: think of doing nil check and then moving to k8s deployment package

Comment (shubham14bajpai) [reply pkg/upgrade/upgrader/helper.go]: Moved the function to deployment package.

**Pull request merged**